Lib/argparse.py

```diff
@@ -1077,7 +1077,7 @@ def __init__(self,
                  prog,
                  parser_class,
                  dest=SUPPRESS,
-                 required=True,
+                 required=False,
```

Comment:
While we are here, please move `required` to the end of the list of parameters. It was inserted in the middle of non-keyword-only parameters.

Reply:
This was discussed [here](https://github.com/python/cpython/pull/3027#discussion_r139233502)

It makes no difference because `add_subparsers` does not take positional arguments (all arguments are keyword-only).  The ordering here was chosen to be consistent with the rest of the classes in the module

Reply:
Oh, sorry. You are right.